### ISSUE TITLE
ManagerBLE.find returns incorrect values

### ISSUE THREAD

**Issue by ThomasHoussin** [athombv/apps-sdk-issues#154]
I run ManagerBLE.find to get updated values for my device (RuuviTag - ble device). I always get the same values in the BleAdvertisement, even when the values send by the device change (confirmed by reading the values in another way).

Here is an extract of the code : 

        Homey.ManagerBLE.find(deviceData.uuid, timeout)
            .then(bleAdv => {
                console.log(bleAdv.rssi);
                this.setCapabilityValue('measure_rssi', bleAdv.rssi);
                return bleAdv.manufacturerData;
            }) 

rssi and manufacturerData do not change ; for example if you move the BLE device away from Homey, RSSI should change but it is not reflected in Homey. In the end sensor values of the RuuviTag reported in Homey  are incorrect because of that. 

Is there a way to force BleAdvertisement refresh ?

**Comment by ThomasHoussin** [comment]
I just checked, v5 (sdk v2) does NOT fix this issue, I still have the same problem.

**Comment by mrlukasbos** [comment]
This is correct. The value returned in `ManagerBLe.find()` does return cached data of the device, which is the data found in the latest 'discovery'. There are two ways to resolve your issue:

1. Connect to the device. 
To get updates on the RSSI you may want to connect to the device. Then you can use the `updateRssi()` function to update the RSSI value. (https://apps-sdk-v3.developer.athom.com/BlePeripheral.html) 

2. Force a rediscover
`ManagerBLe.discover()` forces the BLE advertisements to refresh. You can retrieve your results by finding your advertisement from the discovery results, or by calling `ManagerBLe.find()` again. Because discovery takes some time, this solution is slower than simply connecting to the device.

Please let us know if this solves the issue for you.

**Comment by ThomasHoussin** [comment]
I'm sorry bit it doesn't. It's not possible to connect to the device, I have to use ManagerBle.discover but I still get cached values : 

Below is an example ; in app.js onInit(), I try to do a discover every 30s, and print RSSI, a sequence number (which changes frequently) and the manufacturerData buffer. The lines are always the same (see result). If I restart the app, values are updated, but only once at each restart.  

The code I run : 

```
'use strict';

const Homey = require('homey');
const delay = s => new Promise(resolve => setTimeout(resolve, 1000 * s));

class MyApp extends Homey.App {
  /**
   * onInit is called when the app is initialized.
   */
  async onInit() {
      this.log('MyApp has been initialized');

      while (true) {
          this.homey.ble.discover([], 25000)
              .then(devices => {
                  devices.forEach(device => {
                      //discard all but one device
                      if (device.uuid != 'c014fc8451fc') return;
                      console.log(`RSSI : ${device.rssi}`);
                      console.log(`sequence number : ${device.manufacturerData.readUInt16BE(18)}`);
                      console.log(device.manufacturerData);
                  });
              });
          await delay(30);
      }
  }
}

module.exports = MyApp;
```

And the output : 
 
```
PS > homey app run
✓ Pre-processing app...
✓ Running plugins...
✓ Running plugin `compose`...
[...]
─────────────── Logging stdout & stderr ───────────────
2020-12-15 22:04:50 [log] [MyApp] MyApp has been initialized
RSSI : -99
sequence number : 55812
<Buffer 99 04 05 0e de 45 4a bc 65 00 c0 03 dc 00 40 a9 36 4c da 04 c0 14 fc 84 51 fc>
RSSI : -99
sequence number : 55812
<Buffer 99 04 05 0e de 45 4a bc 65 00 c0 03 dc 00 40 a9 36 4c da 04 c0 14 fc 84 51 fc>
RSSI : -99
sequence number : 55812
<Buffer 99 04 05 0e de 45 4a bc 65 00 c0 03 dc 00 40 a9 36 4c da 04 c0 14 fc 84 51 fc>
RSSI : -99
sequence number : 55812
<Buffer 99 04 05 0e de 45 4a bc 65 00 c0 03 dc 00 40 a9 36 4c da 04 c0 14 fc 84 51 fc>
RSSI : -99
sequence number : 55812
<Buffer 99 04 05 0e de 45 4a bc 65 00 c0 03 dc 00 40 a9 36 4c da 04 c0 14 fc 84 51 fc>
RSSI : -99
sequence number : 55812
<Buffer 99 04 05 0e de 45 4a bc 65 00 c0 03 dc 00 40 a9 36 4c da 04 c0 14 fc 84 51 fc>

--- INFO: com.thomashoussin.ruuvitag has been killed ---
2020-12-15 22:08:18 [log] [MyApp] MyApp has been initialized
RSSI : -95
sequence number : 55978
<Buffer 99 04 05 0e dc 45 45 bc 6c 00 c0 03 dc 00 3c a9 f6 4c da aa c0 14 fc 84 51 fc>
RSSI : -95
sequence number : 55978
<Buffer 99 04 05 0e dc 45 45 bc 6c 00 c0 03 dc 00 3c a9 f6 4c da aa c0 14 fc 84 51 fc>
RSSI : -95
sequence number : 55978
<Buffer 99 04 05 0e dc 45 45 bc 6c 00 c0 03 dc 00 3c a9 f6 4c da aa c0 14 fc 84 51 fc>

--- INFO: com.thomashoussin.ruuvitag has been killed ---
2020-12-15 22:10:13 [log] [MyApp] MyApp has been initialized
RSSI : -101
sequence number : 56062
<Buffer 99 04 05 0e d8 45 44 bc 71 00 c4 03 e4 00 40 a9 36 4c da fe c0 14 fc 84 51 fc>
───────────────────────────────────────────────────────

```

**Comment by ThomasHoussin** [comment]
I found a way to make it work as expected : I had to clear the advertisements with
`delete this.homey.ble.__advertisementsByPeripheralUUID[device.uuid];`

Now I get as a result : 

```
─────────────── Logging stdout & stderr ───────────────
2020-12-15 22:58:55 [log] [MyApp] MyApp has been initialized
RSSI : -98
sequence number : 58341
<Buffer 99 04 05 0e be 45 ec bc 82 00 bc 03 e4 00 44 a9 36 4d e3 e5 c0 14 fc 84 51 fc>
RSSI : -93
sequence number : 58373
<Buffer 99 04 05 0e be 45 e7 bc 81 00 c0 03 d8 00 48 a9 36 4d e4 05 c0 14 fc 84 51 fc>
RSSI : -101
sequence number : 58391
<Buffer 99 04 05 0e be 45 e7 bc 81 00 b8 03 dc 00 38 a9 36 4d e4 17 c0 14 fc 84 51 fc>
```

**Comment by mrlukasbos** [comment]
I see you are correct. This appears to be an issue with our SDK. I have forwarded this issue to our internal issue tracker. Unfortunately I cannot say when the issue will be resolved. I will let you know in this thread when a fix is available. 

I hope for now you can continue working on your app using the workaround you found.

**Comment by mrlukasbos** [comment]
Related: #85 #86

**Comment by mrlukasbos** [comment]
This issue should have been resolved in Homey v6.0.0 which is currently available as an experimental release.
